vpp/oclpostprocess_blender.cpp

```diff
@@ -25,18 +25,182 @@
 
 #include "oclpostprocess_blender.h"
 #include "vaapipostprocess_factory.h"
+#include "common/common_def.h"
 #include "common/log.h"
+#include "ocl/oclcontext.h"
+#include <va/va_drmcommon.h>
 
 namespace YamiMediaCodec{
 
+YamiStatus
+OclPostProcessBlender::blend(const SharedPtr<VideoFrame>& src,
+                             const SharedPtr<VideoFrame>& dst)
+{
+    YamiStatus ret = YAMI_SUCCESS;
+    cl_int cl_status;
+    cl_context context = m_context.get()->m_context;
+    cl_command_queue cmd_q = m_context.get()->m_queue;
+    cl_mem src_cl_mem = NULL;
+    cl_mem dst_cl_mem_y = NULL, dst_cl_mem_uv = NULL;
+    cl_mem bg_cl_mem_y = NULL, bg_cl_mem_uv = NULL;
+    cl_import_image_info_intel import_info;
+    size_t global_work_size[2], local_work_size[2];
+    uint32_t crop_x, crop_y, crop_w, crop_h;
+    uint32_t element_size;
+
+    VASurfaceID src_id = (VASurfaceID)src->surface;
+    VASurfaceID dst_id = (VASurfaceID)dst->surface;
+    VAImage src_image, dst_image;
+    VABufferInfo src_info, dst_info;
+
+    if (VA_STATUS_SUCCESS != vaDeriveImage(m_display, src_id, &src_image)) {
```

Comment:
could we use one function like 
 cl_mem convertToClImage(const SharedPtr<VideoFrame>& frame, cl_image_format format, int plane)  to hide all vaDeriveImage, vaAcquireBufferHandle, clCreateImageFromFdINTEL, logical